### ISSUE TITLE
bug with "get_nns_by_item"?

### ISSUE THREAD

**Issue by Horace89** [spotify/annoy#114]
Hi. I think I have uncovered a bug in "get_nns_by_item". Here is my code to reproduce it.
The bug does not seem to appear in "get_nns_by_vector".

```
import random


import annoy
DIM = 50
t = annoy.AnnoyIndex(DIM, metric='angular')  # Length of item vector that will be indexed

for i in range(100):
    v = [random.gauss(0, 1) for _ in range(DIM)]
    t.add_item(i, v)

t.build(10)

print('nb_items:', t.get_n_items())

#BUG Annoy : TypeError: get_nns_by_item() takes no keyword arguments

#works...
#res1 = t.get_nns_by_item(50, 3, -1, True)
#does not work:
res1 = t.get_nns_by_item(50, 3, search_k=-1, include_distances=True)
print('res1:', res1)

#both calls work with get_nns_by_vector:
vq = [random.gauss(0, 1) for _ in range(DIM)]
res2a = t.get_nns_by_vector(vq, 2, -1, True)
res2b = t.get_nns_by_vector(vq, 2, search_k=-1, include_distances=True)
print('res2a:', res2a)
print('res2b:', res2b)
```

**Comment by erikbern** [comment]
Did you try the latest code? I think I actually fixed this yesterday

https://github.com/spotify/annoy/commit/ae74624fcdc89db6ab947149b7db7545212bb062

**Comment by erikbern** [comment]
The underlying python C extension doesn't handle arguments so well but I solved it by a dumb wrapper on the Python side

**Comment by Horace89** [comment]
ok, I'll test it again tomorrow. Thank you.

**Comment by Horace89** [comment]
Yes, I confirm that I use the latest version of Annoy (1.5.2), and the described bug still occurs. I'm on Python 3.4.3.

**Comment by erikbern** [comment]
When I meant latest version I meant bleeding edge github master.

I updated master but didn't push it to pypi. I can bump the version and
push. I keep forgetting to do that so in glad you reminded me :)

**Comment by erikbern** [comment]
I bumped the version to 1.6.0 and pushed it to PyPI

**Comment by dudevil** [comment]
@erikbern 
I'm having trouble installing annoy through pip after this update:
```
Installing collected packages: annoy
  Found existing installation: annoy 1.5.2
    Uninstalling annoy:
      Successfully uninstalled annoy
  Running setup.py install for annoy
    building 'annoy.annoylib' extension
    x86_64-linux-gnu-gcc -pthread -fno-strict-aliasing -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -fPIC -I/usr/include/python2.7 -c src/annoymodule.cc -o build/temp.linux-x86_64-2.7/src/annoymodule.o -O3 -march=native -ffast-math
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    src/annoymodule.cc:16:24: fatal error: kissrandom.h: No such file or directory
     #include "kissrandom.h"
                            ^
    compilation terminated.
    error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
```
Clonning and running `python setup.py install` works fine.

**Comment by erikbern** [comment]
let me take a look at it

**Comment by erikbern** [comment]
I had forgotten to include src/kissrandom.h in setup.py.

I added it and pushed 1.6.1.

Sorry about the inconvenience

**Comment by erikbern** [comment]
hm still not working – not sure what's going on. let me try again

**Comment by erikbern** [comment]
try again – had to bump to 1.6.2 :(

**Comment by dudevil** [comment]
Works great now. Thanks for the quick fix!

**Comment by piskvorky** [comment]
That's what I call rapid development! :)

Btw PyPI treats `rc1`, `rc2` tags as proper releases, which pip will ignore when users ask to install (a good option for release checking).

**Comment by Horace89** [comment]
It's working great now! Thanks!
